packages/libpostal/src/parse-address-with.js

```diff
@@ -1,36 +1,86 @@
-import postal from '@cymen/node-postal';
-import { get, set, clone } from 'lodash';
-
-const parse = (input) => ({
-    id: input,
-    value: postal.parser
-        .parse_address(String(input).trim())
-        .reduce((obj, cur) => ({ ...obj, [cur.component]: cur.value }), {}),
-});
+import { get } from 'lodash';
+import parse from './postal/parse';
 
 /**
- * Takes a field of object containing an address to return the same object except for the field containing the address.
- * This will contain the different fields found in the address.
- *
- * @name parseAddressWith
- * @param {String} [path] path to the chosen field
- * @returns {Object}
+ * ParseAddressWith function see documentation at the end.
+ * This part of the doc is used for jsdoc typing
+ * @private
+ * @param data {unknown}
+ * @param feed {Feed}
+ * @param ctx {import('../../core/src/engine').EngineScope}
  */
-
-export default function parseAddressWith(data, feed) {
+const parseAddressWith = (data, feed, ctx) => {
     const paths = []
-        .concat(this.getParam('path'))
+        .concat(ctx.getParam('path'))
         .filter(Boolean);
-    if (this.isLast()) {
+    if (ctx.isLast()) {
         return feed.close();
     }
-    const tada = clone(data);
+
     paths.forEach((path) => {
         const value = get(data, path);
         if (Array.isArray(value)) {
-            return set(tada, path, value.map(parse));
+            return feed.send(value.map(entry => parse(entry, path)));
+        }
+        if (typeof value === 'string') {
+            return feed.send(parse(value, path));
         }
-        return set(tada, path, parse(value));
+        return feed.send(value);
     });
-    return feed.send(tada);
-}
+    return feed.end();
+};
+
+/**
+ * Try to parse given addresss.
+ *
+ * Essayer de faire l'analyse grammaticale des adresses données.
+ *
+ * ### Example / Exemple
+ *
+ * #### Script / Scénario
+ *
+ * ```ini
+ * ; Import libpostal plugin required to use "parseAddressWith"
+ * ; Importation du plugin libpostal nécessaire pour utiliser "parseAddressWith"
+ * [use]
+ * plugin = libpostal
+ *
+ * ; Using "parseAddressWith"
+ * ; Utilisation de "parseAddressWith"
+ * [expandAddress]
+ * ; path = value
+ *
+ * ```
+ *
+ * #### Input / Entrée
+ *
+ * ```json
+ *  [
+ *      {
+ *          "value": "Barboncino 781 Franklin Ave, Crown Heights, Brooklyn, NY 11238"
+ *      }
+ *  ]
+ * ```
+ *
+ * #### Output / Sortie
+ *
+ * ```json
+ *  [
+ *      {
+ *          "id": "Barboncino 781 Franklin Ave, Crown Heights, Brooklyn, NY 11238",
+ *          "value": [
+ *              "barboncino 781 franklin avenue crown heights brooklyn ny 11238",
+ *              "barboncino 781 franklin avenue crown heights brooklyn new york 11238"
+ *          ]
+ *      }
+ *  ]
+ * ```
+ *
+ * @name expandAddressWith
```

Comment:
```suggestion
 * @name parseAddressWith
```

I assume all the doc has to be rewritten too.